src/rules/require-hook.ts

```diff
@@ -39,9 +39,11 @@ const shouldBeInHook = (
     case AST_NODE_TYPES.ExpressionStatement:
       return shouldBeInHook(node.expression, context, allowedFunctionCalls);
     case AST_NODE_TYPES.CallExpression:
+      const nodeName = getNodeName(node);
```

Comment:
This is a runtime de-optimization - previously we'd only determine the node name if the other condition hasn't matched; now we'll always be getting the node name even if we don't need it